google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1577,4 +1586,170 @@ public URL generateV4GPutbjectSignedUrl(String bucketName, String objectName)
+    HmacKeyMetadata metadata =
+        storage.getHmacKey(accessId, Storage.GetHmacKeyOption.projectId(projectId));
+    HmacKeyMetadata newMetadata = storage.updateHmacKeyState(metadata, HmacKeyState.ACTIVE);
+
+    System.out.println("The HMAC key is now active.");
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("ID: " + newMetadata.getId());
+    System.out.println("Access ID: " + newMetadata.getAccessId());
+    System.out.println("Project ID: " + newMetadata.getProjectId());
+    System.out.println("Service Account Email: " + newMetadata.getServiceAccount().getEmail());
+    System.out.println("State: " + newMetadata.getState().toString());
+    System.out.println("Time Created: " + new Date(newMetadata.getCreateTime()).toString());
+    System.out.println("Time Updated: " + new Date(newMetadata.getUpdateTime()).toString());
+    System.out.println("ETag: " + newMetadata.getEtag());
+    // [END storage_activate_hmac_key]
+    return newMetadata;
+  }
+
+  /** Example of deactivating an HMAC key. * */
+  public HmacKeyMetadata deactivateHmacKey(String accessId, String projectId)
+      throws StorageException {
+    // [START storage_deactivate_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The access ID of the HMAC key, e.g. "GOOG0234230X00"
+    // String accessId = "GOOG0234230X00";
+    //
+    // The ID of the project to which the service account belongs.
+    // String projectId = "project-id";
+    HmacKeyMetadata metadata =
+        storage.getHmacKey(accessId, Storage.GetHmacKeyOption.projectId(projectId));
+    HmacKeyMetadata newMetadata = storage.updateHmacKeyState(metadata, HmacKeyState.INACTIVE);
+
+    System.out.println("The HMAC key is now inactive.");
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("ID: " + newMetadata.getId());
+    System.out.println("Access ID: " + newMetadata.getAccessId());
+    System.out.println("Project ID: " + newMetadata.getProjectId());
+    System.out.println("Service Account Email: " + newMetadata.getServiceAccount().getEmail());
+    System.out.println("State: " + newMetadata.getState().toString());
+    System.out.println("Time Created: " + new Date(newMetadata.getCreateTime()).toString());
+    System.out.println("Time Updated: " + new Date(newMetadata.getUpdateTime()).toString());
+    System.out.println("ETag: " + newMetadata.getEtag());
+    // [END storage_deactivate_hmac_key]
+    return newMetadata;
+  }
+
+  public Page<HmacKeyMetadata> listHmacKeys(String serviceAccountEmail, String projectId)
+      throws StorageException {
+    // [START storage_list_hmac_keys]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The service account to list HMAC keys for.
+    // String serviceAccountEmail = "service-account@iam.gserviceaccount.com";
```

Comment:
This deviates a bit from other samples.

Reply:
in what way?

Reply:
Node.js: https://github.com/googleapis/nodejs-storage/blob/master/samples/hmacKeysList.js#L24
Ruby: https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/storage/hmac.rb#L15
Python: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/storage/cloud-client/hmac.py#L22

Service account parameter isn't set.

Reply:
Gotcha!